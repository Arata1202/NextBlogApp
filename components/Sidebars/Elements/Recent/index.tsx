'use client';

import React from 'react';
import { useTheme } from 'next-themes';
import { BellAlertIcon } from '@heroicons/react/24/solid';
import { Article } from '@/libs/microcms';
import SidebarArticleListItem from '../Elements/SidebarArticleListItem';

type Props = {
  articles: Article[];
};

export default function Recent({ articles }: Props) {
  const { theme } = useTheme();

  const sortedArticles = articles
    ?.slice()
    .sort((a, b) => {
      const dateA = a.publishedAt ? new Date(a.publishedAt).getTime() : 0;
      const dateB = b.publishedAt ? new Date(b.publishedAt).getTime() : 0;
      return dateB - dateA;
    })
    .slice(0, 3);

  return (
    <>
      <div
        className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <div className={`text-2xl text-center font-semibold flex justify-center`}>
          <BellAlertIcon className="h-8 w-8 mr-2" />
          最新の投稿
        </div>
        {sortedArticles.map((article) => (
          <SidebarArticleListItem key={article.id} article={article} />
        ))}
      </div>
    </>
  );
}
