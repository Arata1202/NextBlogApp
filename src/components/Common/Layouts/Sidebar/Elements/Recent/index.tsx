'use client';

import { useTheme } from 'next-themes';
import { BellAlertIcon } from '@heroicons/react/24/solid';
import { Article } from '@/libs/microcms';
import styles from './index.module.css';

type Props = {
  recentArticles: Article[];
};

export default function Recent({ recentArticles }: Props) {
  const { theme } = useTheme();

  const sortedArticles = recentArticles
    ?.slice()
    .sort((a, b) => {
      const dateA = a.publishedAt ? new Date(a.publishedAt).getTime() : 0;
      const dateB = b.publishedAt ? new Date(b.publishedAt).getTime() : 0;
      return dateB - dateA;
    })
    .slice(0, 3);

  return (
    <>
      <div
        className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <div className={`text-2xl text-center font-semibold flex justify-center`}>
          <BellAlertIcon className="h-8 w-8 mr-2" />
          最新の投稿
        </div>
        {sortedArticles.map((article) => (
          <ul
            key={article.id}
            className={`border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
          >
            <li>
              <a href={`/articles/${article.id}`} className={styles.link}>
                <img src={article.thumbnail.url} alt="サムネイル" className={styles.image} />
                <div className={`${styles.title} font-bold`}>{article.title}</div>
              </a>
            </li>
          </ul>
        ))}
      </div>
    </>
  );
}
