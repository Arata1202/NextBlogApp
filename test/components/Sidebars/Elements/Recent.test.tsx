import { render } from '@testing-library/react';
import Recent from '@/components/Sidebars/Elements/Recent';

jest.mock('@/components/Sidebars/Elements/Elements/SidebarArticleListItem');

describe('Recent', () => {
  test('スナップショット（Recent）', () => {
    const { asFragment } = render(<Recent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
