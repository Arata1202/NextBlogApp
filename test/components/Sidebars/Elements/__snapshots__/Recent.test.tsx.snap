// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent スナップショット（Recent） 1`] = `
<DocumentFragment>
  <div
    class="bg-white pt-8 px-4 border border-gray-300 py-5 mt-5"
  >
    <h1
      class="profile text-2xl text-center font-semibold flex justify-center"
    >
      <svg
        aria-hidden="true"
        class="h-8 w-8 mr-2"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.85 3.5a.75.75 0 0 0-1.117-1 9.719 9.719 0 0 0-2.348 4.876.75.75 0 0 0 1.479.248A8.219 8.219 0 0 1 5.85 3.5ZM19.267 2.5a.75.75 0 1 0-1.118 1 8.22 8.22 0 0 1 1.987 4.124.75.75 0 0 0 1.48-.248A9.72 9.72 0 0 0 19.266 2.5Z"
        />
        <path
          clip-rule="evenodd"
          d="M12 2.25A6.75 6.75 0 0 0 5.25 9v.75a8.217 8.217 0 0 1-2.119 5.52.75.75 0 0 0 .298 1.206c1.544.57 3.16.99 4.831 1.243a3.75 3.75 0 1 0 7.48 0 24.583 24.583 0 0 0 4.83-1.244.75.75 0 0 0 .298-1.205 8.217 8.217 0 0 1-2.118-5.52V9A6.75 6.75 0 0 0 12 2.25ZM9.75 18c0-.034 0-.067.002-.1a25.05 25.05 0 0 0 4.496 0l.002.1a2.25 2.25 0 1 1-4.5 0Z"
          fill-rule="evenodd"
        />
      </svg>
      最新の投稿
    </h1>
    <p
      class="text-center"
    >
      新着記事はありません
    </p>
  </div>
</DocumentFragment>
`;
